src/rules.c

```diff
@@ -19,8 +21,11 @@ void rule_apply(rule_t * r, notification * n)
                 n->allow_markup = r->allow_markup;
         if (r->plain_text != -1)
                 n->plain_text = r->plain_text;
-        if (r->new_icon)
-                n->icon = r->new_icon;
+        if (r->new_icon) {
+                if(n->icon)
+                        free(n->icon);
```

Comment:
Since we already include glib, let's avoid unecessary includes and use g_free here.